core/imt/inc/ROOT/TExecutor.hxx

```diff
@@ -0,0 +1,494 @@
+/// deletes what it returns, it simply forgets it.\n
+///
+/// \param func
+/// \parblock
+/// a lambda expression, an std::function, a loaded macro, a
+/// functor class or a function that takes zero arguments (for the first signature)
+/// or one (for the second signature).
+/// \endparblock
+/// \param args
+/// \parblock
+/// a standard vector, a ROOT::TSeq of integer type or an initializer list for the second signature.
+/// An integer only for the first.\n
+/// \endparblock
+///
+/// **Note:** in cases where the function to be executed takes more than
+/// zero/one argument but all are fixed except zero/one, the function can be wrapped
+/// in a lambda or via std::bind to give it the right signature.\n
+///
+/// #### Return value:
+/// An std::vector. The elements in the container
+/// will be the objects returned by func.
+///
+/// ### ROOT::Internal::TExecutor::MapReduce
+/// This set of methods behaves exactly like Map, but takes an additional
+/// function as a third argument. This function is applied to the set of
+/// objects returned by the corresponding Map execution to "squash" them
+/// to a single object. This function should be independent of the size of
+/// the vector returned by Map due to optimization of the number of chunks.
+///
+/// #### Examples:
+/// ~~~{.cpp}
+/// root[] ROOT::Internal::TExecutor pool; auto ten = pool.MapReduce([]() { return 1; }, 10, [](std::vector<int> v) { return std::accumulate(v.begin(), v.end(), 0); })
+/// root[] ROOT::Internal::TExecutor pool(ROOT::Internal::ExecutionPolicy::kMultiprocess); auto hist = pool.MapReduce(CreateAndFillHists, 10, PoolUtils::ReduceObjects);
+/// ~~~
+///
+//////////////////////////////////////////////////////////////////////////
+
+
+namespace ROOT{
+
+namespace Internal{
+class TExecutor: public TExecutorCRTP<TExecutor> {
+public:
+
+   /// \brief Class constructor. Sets the default execution policy and initializes the corresponding executor.
+   /// Defaults to multithreaded execution policy if ROOT is compiled with IMT=ON and IsImplicitMTEnabled. Otherwise it defaults to a serial execution policy
+   /// \param nProcessingUnits [optional] Number of parallel processing units, only taken into account if the execution policy is kMultithread
+   explicit TExecutor(unsigned nProcessingUnits = 0) :
+      TExecutor(ROOT::IsImplicitMTEnabled() ? ROOT::Internal::ExecutionPolicy::kMultithread : ROOT::Internal::ExecutionPolicy::kSerial, nProcessingUnits) {}
+
+   /// \brief Class constructor. Sets the execution policy and initializes the corresponding executor.
+   /// \param execPolicy Execution policy(kMultithread, kMultiprocess, kSerial) to process the data
+   /// \param nProcessingUnits [optional] Number of parallel processing units, only taken into account if the execution policy is kMultithread
+   explicit TExecutor(ROOT::Internal::ExecutionPolicy execPolicy, unsigned nProcessingUnits = 0) : fExecPolicy(execPolicy) {
+      fExecPolicy = execPolicy;
```

Comment:
Why is this needed?

Reply:
This is the only constructor that allows you to pass an explicit execution policy